File: Data Structures & Algorithms/design-hashmap/submission-0.java
class MyHashMap {
    int[] keys;
    int[] vals;
    public MyHashMap() {
        keys = new int[1000001];
        vals = new int[1000001];
    }
    
    public void put(int key, int value) {
        keys[key] = 1;
        vals[key] = value;
    }
    
    public int get(int key) {
        if (keys[key] == 0) {
            return -1;
        }
        return vals[key];
    }
    
    public void remove(int key) {
        keys[key] = 0;
    }
}

/**
 * Your MyHashMap object will be instantiated and called as such:
 * MyHashMap obj = new MyHashMap();
 * obj.put(key,value);
 * int param_2 = obj.get(key);
 * obj.remove(key);
 */

File: Data Structures & Algorithms/range-sum-query-2d-immutable/submission-0.cs
public class NumMatrix {
    private int[][] prefix;

    public NumMatrix(int[][] matrix) {
        if (matrix == null || matrix.Length == 0 || matrix[0].Length == 0) {
            return;
        }
        
        int rows = matrix.Length;
        int cols = matrix[0].Length;
        
        prefix = new int[rows + 1][];
        for (int i = 0; i <= rows; i++) {
            prefix[i] = new int[cols + 1];
        }
        for (int r = 0; r < rows; r++) {
            for (int c = 0; c < cols; c++) {
                prefix[r + 1][c + 1] = matrix[r][c] + prefix[r][c + 1] + prefix[r + 1][c] - prefix[r][c];
            }
        }
    }
    
    public int SumRegion(int row1, int col1, int row2, int col2) {
        return prefix[row2 + 1][col2 + 1] - prefix[row1][col2 + 1] - prefix[row2 + 1][col1] + prefix[row1][col1];
    }
}

/**
 * Your NumMatrix object will be instantiated and called as such:
 * NumMatrix obj = new NumMatrix(matrix);
 * int param_1 = obj.SumRegion(row1,col1,row2,col2);
 */